jaxrs-api/src/test/java/jakarta/ws/rs/core/CookieTest.java

```diff
@@ -37,24 +38,80 @@ public void tearDown() throws Exception {
         RuntimeDelegate.setInstance(null);
     }
 
-    /**
-     * Test of equals method, of class Cookie and NewCookie.
-     */
     @Test
-    public void testEquals() {
-        Object nullObj = null;
+    public final void shouldReturnFalseWhenComparingCookieToNullObject() {
+        Cookie cookie = new Cookie("name", "value");
+        assertFalse(cookie.equals(null));
+
+        cookie = new Cookie.Builder("name").value("value").build();
+        assertFalse(cookie.equals(null));
+    }
+
+    @Test
+    public final void shouldReturnFalseWhenComparingCookieToNewCookie() {
         Cookie cookie = new Cookie("name", "value");
-        Cookie cookie1 = new Cookie("name", "value");
-        Cookie cookie2 = new Cookie("name", "value2");
         NewCookie newCookie = new NewCookie("name", "value");
-        NewCookie newCookie1 = new NewCookie("name", "value");
-        NewCookie newCookie2 = new NewCookie("name", "value2");
-        assertFalse(cookie.equals(nullObj));
         assertFalse(cookie.equals(newCookie));
+
+        Cookie thisCookie = new Cookie.Builder("name").value("value").build();
+        NewCookie thatNewCookie = new NewCookie.Builder("name").value("value").build();
+        assertFalse(thisCookie.equals(thatNewCookie));
+    }
+
+    @Test
+    public final void shouldReturnFalseWhenComparingCookiesThatHaveDifferentValues() {
+        Cookie cookie = new Cookie("name", "value");
+        Cookie cookie2 = new Cookie("name", "value2");
         assertFalse(cookie.equals(cookie2));
+
+        Cookie thisCookie = new Cookie.Builder("name").value("value").build();
+        Cookie thatCookie = new Cookie.Builder("name").value("value2").build();
+        assertFalse(thisCookie.equals(thatCookie));
+    }
+
+    @Test
+    public final void shouldReturnTrueWhenComparingCookiesThatHaveSameValues() {
+
+        Cookie cookie = new Cookie("name", "value");
+        Cookie cookie1 = new Cookie("name", "value");
+        NewCookie newCookie = new NewCookie("name", "value");
         assertTrue(cookie.equals(cookie1));
         assertTrue(cookie.equals(newCookie.toCookie()));
-        assertTrue(newCookie.equals(newCookie1));
-        assertFalse(newCookie.equals(newCookie2));
+
+        Cookie thisCookie = new Cookie.Builder("name").value("value").build();
+        Cookie thatCookie = new Cookie.Builder("name").value("value").build();
+        assertTrue(thisCookie.equals(thatCookie));
+
+        thatCookie = new NewCookie.Builder("name").value("value").build().toCookie();
+        assertTrue(thisCookie.equals(thatCookie));
     }
+
+    @Test
+    public final void shouldThrowAnIllegalArgumentExceptionWhenBuildingCookieWithNullName() {
+
+        try {
+            new Cookie(null, "value");
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+
+        try {
+			new Cookie(null, "value", "path", "domain");
```

Comment:
```suggestion
		new Cookie(null, "value", "path", "domain");
```

extra whitespace here - PMD or CheckStyle may complain.

Reply:
@NicoNes To finally merge your contribution to `master` I'd kindly like to ask the following of you:
* Please fix the above whitespace issue reported by @andymc12 (to prevent PMD / CheckStyle complaints).
* Rebase the whole PR ontop `master` to fix the CI fail on JDK 14+ (should be fixed by `master` already).

I will be happy to merge your change into `master` then.

Reply:
@NicoNes Oh, and if you like, feel free to also mention it in https://github.com/eclipse-ee4j/jaxrs-api/blob/master/jaxrs-spec/src/main/asciidoc/chapters/appendix/_changes-since-3.0-release.adoc as part of the same PR. Thanks. :-)

Reply:
@mkarg and @andymc12 I made the fix and the rebase into `master` and I change the target branch of this PR to `master`.
I also mention the changes into the release document. Please feel free to correct my wording if necessary.

Thanks